File: app/src/main/java/com/sorcerer/sorcery/iconpack/models/AppfilterItem.java
package com.sorcerer.sorcery.iconpack.models;

/**
 * @description:
 * @author: Sorcerer
 * @date: 2016/11/29
 */

public class AppfilterItem {
    private String mComponent;
    private String mDrawable;

    public AppfilterItem(String component, String drawable) {
        mComponent = component;
        mDrawable = drawable;
    }

    public String getComponent() {
        return mComponent;
    }

    public void setComponent(String component) {
        mComponent = component;
    }

    public String getDrawable() {
        return mDrawable;
    }

    public void setDrawable(String drawable) {
        mDrawable = drawable;
    }

    public String getPackageName() {
        return mComponent.split("/")[0];
    }
}

File: app/src/main/java/com/sorcerer/sorcery/iconpack/data/models/AppfilterItem.kt
package com.sorcerer.sorcery.iconpack.data.models

/**
 * @description:
 * *
 * @author: Sorcerer
 * *
 * @date: 2016/11/29
 */

class AppfilterItem(var component: String, var drawable: String) {
    val packageName: String
        get() = component.split("/".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()[0]
}
